lib/resources/directory.rb

```diff
@@ -21,5 +21,11 @@ def exist?
     def to_s
       "Directory #{source_path}"
     end
+
+    def file_count
+      return unless file.exist?
+      # intentionally used entries as we don't want to recurse/glob
+      Dir.entries(file).reject { |f| f == '.' || f == '..' }.count
```

Comment:
Using a Dir.entries will only work on the local transport. Even then I am getting errors using this like:
```
(master)> _inspec exec file_count.rb

Profile: tests from file_count.rb (tests from file_count.rb)
Version: (not specified)
Target:  local://

  Directory /tmp
     ✔  should exist
     ✔  should be directory
     ×  file_count
     no implicit conversion of Train::File::Local::Unix into String
```
Changing it to `file.path` fixes the issue for local but will cause unexpected results when using a remote target. I think we need to add more logic into train if we want to do this: https://github.com/inspec/train/tree/master/lib/train/file. Adding a Directory class in Train would be a good start i think.